### PR TITLE
Add support for murmur-hash and make it default

### LINKED ISSUE #22: Add MurmurHash3 alongside CRC32
MASH is using MurmurHash3

### LINKED ISSUE #23: Call CRC32 2 times for k>16
in general, support k>16; this would also need a new template type

### PR DESCRIPTION
**Pull request by danieldanciu**

### CODE REVIEW

File: sketch/hash_base.hpp [hunk 1]
```diff
@@ -74,9 +81,14 @@ class HashBase : public SketchBase<std::vector<T>, true> {
                 return val;
             }
             case HashAlgorithm::crc32: {
-                T val = _mm_crc32_u32((uint32_t)crc32_base, (uint32_t)key);
+                uint32_t val = _mm_crc32_u32(hash_seed, (uint32_t)key);
                 return _mm_crc32_u32((uint32_t)val, (uint32_t)index);
             }
+            case HashAlgorithm::murmur:
+                uint32_t val;
+                MurmurHash3_x86_32(&key, sizeof(key), hash_seed, &val);
```
Comment (RagnarGrootKoerkamp) [inline sketch/hash_base.hpp]: Here is what MASH does:
https://github.com/marbl/Mash/blob/master/src/mash/hash.cpp#L14-L24

Seems like they use one call to the `_128` variant on 64bit architectures, and then only use half of the output bits.

Maybe `key` and `index` can also be combined into 1 to only do one call?

Comment (ajoudaki) [inline sketch/hash_base.hpp]: looks like the base function `MurmurHash3_x867_32` works on a 32bit input, while `hash` takes `size_t` input, which could potentially be 64 bits, I suggest to overload the hash function with two separate input types and call the appropriate base hash accordingly?

**Comment by danieldanciu** [comment]
Hmm, then maybe we have to just remove the template parameter from the hash
functions and always assume 64 bits?

**Comment by danieldanciu** [comment]
Actually, we can probably keep the T param, and just return the first
sizeof(T) bytes from the 128-bit hash value. Not sure if this is kosher -
i.e. if the bits in the hash are uniformly distributed, but, oh well.

**Comment by ajoudaki** [comment]
I think checking sizeof and applying the method accordingly is ok.

File: sketch/hash_base.hpp (hunk 1 above)
Comment (danieldanciu) [reply sketch/hash_base.hpp]: I tried to emulate what they do, and also called crc32 twice with two different seeds for 64 bit types.

**Review by RagnarGrootKoerkamp — APPROVED** [review]

**Pull request merged**